src/AutocompleteBadges.js

```diff
@@ -0,0 +1,145 @@
+'use strict';
+
+import templates from './AutocompleteBadges.soy.js';
+import Autocomplete from 'metal-autocomplete';
+import Component from 'metal-component';
+import Soy from 'metal-soy';
+
+class AutocompleteBadges extends Component {
+
+	/**
+	 * @inheritDoc
+	 */
+	attached() {
+		this.inputElement = this.element.querySelector('.autocomplete-field');
+		this.listBadgesElement = this.element.querySelector('.autocomplete-badges--list');
+
+		this.autocomplete_ = new Autocomplete({
+			inputElement: this.inputElement,
+			data: this.getFilteredElements_.bind(this)
+		});		
+
+		this.autocomplete_.on('select', this.onListItemSelected_.bind(this));
+
+		this.element.style.width = this.inputElement.offsetWidth + 'px';
```

Comment:
Are you sure this is necessary? `this.element` is already the parent of `this.inputElement`, so it will already have the appropriate height.